plenum/server/request_handlers/txn_author_agreement_handler.py

```diff
@@ -60,3 +56,22 @@ def _decode_state_value(self, encoded):
                                                                       serializer=config_state_serializer)
             return value, last_seq_no, last_update_time
         return None, None, None
+
+    def _validate_add_taa(self, request):
+        if request.operation.get(TXN_AUTHOR_AGREEMENT_RETIRED, False):
+            raise InvalidClientRequest(request.identifier, request.reqId,
+                                       "Cannot create a transaction author agreement with a 'retired' field.")
+
+    def _validate_update_taa(self, request, digest):
+        # check TAA text
+        text = request.operation.get(TXN_AUTHOR_AGREEMENT_TEXT)
+        ledger_taa = self.get_from_state(StaticTAAHelper.state_path_taa_digest(digest))[0]
+        if ledger_taa.get(TXN_AUTHOR_AGREEMENT_TEXT, text) != text:
```

Comment:
If text is not supplied in request transaction will be rejected, however AFAIK we need to support updating retirement date with just TAA version, without supplying full text.